classes/class3/hw/routes/cat.js

```diff
@@ -0,0 +1,74 @@
+var path = require('path');
+var Cat = require(path.join(__dirname,'../models/catmodel'));
+
+var namesList = ['Indycule','Ignatz','Hume','Dunka','Brockie','Julie','Opie','Chaucer','Lew','Spamela','Tumbleweed'];
+var colorList = ['red','brown','green','starbuck','blue','pink','white','gray'];
+
+var generateName = function () {
+	return namesList[Math.floor(Math.random() * namesList.length)]
+}
+
+var generateRandomAge = function (){
+	return Math.ceil(20*Math.random())
+}
+
+var generateListofColors = function (){
+	var colorArray = [];
+
+	for (var i = 0; i < (colorList.length)-2; i++){
+		colorArray.push(colorList[Math.floor(Math.random() * colorList.length)])
+	}
+
+	return colorArray;
+}
+
+var catsNew = function(req,res){
+	
+	var catX = new Cat({
+		name: generateName(), 
+		age: generateRandomAge(), 
+		colorlist: generateListofColors()
+	});
+	console.log(catX)
+	catX.save(function (err) {
+	  if (err) {
+	    console.log("Problem saving bob", err);
```

Comment:
Every cat probably isn't named "bob". Including information about the data that you tried to save is a good idea, though.